spark/src/main/scala/geotrellis/spark/TileLayerMetadata.scala

```diff
@@ -157,20 +157,43 @@ object TileLayerMetadata {
     K: (? => TilerKeyMethods[K, K2]) ,
     V <: CellGrid,
     K2: SpatialComponent: Boundable
-  ](rdd: RDD[(K, V)], crs: CRS, scheme: LayoutScheme): (Int, TileLayerMetadata[K2]) = {
+  ](rdd: RDD[(K, V)], crs: CRS, scheme: LayoutScheme, maxZoom: Option[Int] = None): (Int, TileLayerMetadata[K2]) = {
     val (extent, cellType, cellSize, bounds) = collectMetadata(rdd)
-    val LayoutLevel(zoom, layout) = scheme.levelFor(extent, cellSize)
+    val LayoutLevel(zoom, layout) = {
+      maxZoom match {
+        case Some(zoom) =>
+          scheme match {
+            case zoomedLayoutScheme: ZoomedLayoutScheme =>
+              zoomedLayoutScheme.levelForZoom(maxZoom.get)
+            case _ => throw new RuntimeException("ZoomedLayoutScheme required when setting a max zoom level")
+          }
+        case None =>
+          scheme.levelFor(extent, cellSize)
+      }
+    }
     val kb = bounds.setSpatialBounds(KeyBounds(layout.mapTransform(extent)))
     (zoom, TileLayerMetadata(cellType, layout, extent, crs, kb))
   }
 
+
   def fromRdd[
     K: GetComponent[?, ProjectedExtent]: (? => TilerKeyMethods[K, K2]),
     V <: CellGrid,
     K2: SpatialComponent: Boundable
-  ](rdd: RDD[(K, V)], scheme: LayoutScheme): (Int, TileLayerMetadata[K2]) = {
+  ](rdd: RDD[(K, V)], scheme: LayoutScheme, maxZoom: Option[Int] = None): (Int, TileLayerMetadata[K2]) = {
     val (extent, cellType, cellSize, bounds, crs) = collectMetadataWithCRS(rdd)
-    val LayoutLevel(zoom, layout) = scheme.levelFor(extent, cellSize)
+    val LayoutLevel(zoom, layout) = {
+      maxZoom match {
+        case Some(zoom) =>
+          scheme match {
+            case zoomedLayoutScheme: ZoomedLayoutScheme =>
+              zoomedLayoutScheme.levelForZoom(maxZoom.get)
+            case _ => throw new RuntimeException("ZoomedLayoutScheme required when setting a max zoom level")
```

Comment:
Instead of throwing here, I would like to enforce this with the type system. This is a situation where I would prefer some code duplication for type safety. So keeping the original fromRDD's, and adding

```scala
def fromRdd[
    K: GetComponent[?, ProjectedExtent]: (? => TilerKeyMethods[K, K2]),
    V <: CellGrid,
    K2: SpatialComponent: Boundable
  ](rdd: RDD[(K, V)], scheme: ZoomedLayoutScheme, maxZoom: Int): (Int, TileLayerMetadata[K2]) = {
    val (extent, cellType, cellSize, bounds, crs) = collectMetadataWithCRS(rdd)
    val LayoutLevel(zoom, layout) =
      zoomedLayoutScheme.levelForZoom(maxZoom)

    val GridBounds(colMin, rowMin, colMax, rowMax) = layout.mapTransform(extent)
    val kb = bounds.setSpatialBounds(KeyBounds(layout.mapTransform(extent)))
    (zoom, TileLayerMetadata(cellType, layout, extent, crs, kb))
  }
```

(and one for the other type of `fromRdd` call.